.github/workflows/ci.yml

```
@@ -0,0 +1,265 @@
+name: CI - Build, Test, and Push Docker Image
+
+on:
+  push:
+    branches: master
+    tags: 'v*'
+  pull_request:
+    branches: master
+
+env:
+  IMAGE_NAME: hazelcast/simulator
+  PYTHON_VERSION: "3.11"
+  DOCKER_PLATFORMS: linux/amd64,linux/arm64
+  LOCAL_REGISTRY_IMAGE: localhost:5000/hazelcast/simulator:test
+
+jobs:
+  docker-ci:
+    name: Docker Build, Test, and Push
+    runs-on: ubuntu-latest
+    services:
+      # Local Docker registry for testing multi-platform images
+      registry:
+        image: registry:2
+        ports:
+          - 5000:5000
+        options: >-
+          --health-cmd "wget --no-verbose --tries=1 --spider http://localhost:5000/ || exit 1"
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3.11.1
```

```suggestion
        uses: docker/setup-buildx-action@v3
```

Nit use floating tag elsewhere

https://github.com/hazelcast/hazelcast-docker/blob/9466756692553823bfca159b28e007e27cf06730/.github/actions/setup-docker/action.yml#L9

Also applied for other `docker` actions

---

changed all tag references to floating tag